### PR TITLE
[#111428758] Add a container certstrap CA manager tool

### PR DESCRIPTION
**Pull request by keymon**
[Auto-generate SSLs for all internal components](https://www.pivotaltracker.com/story/show/111428758)

What
----

We want to generate the certificates for all the components, signed by a CA which is trusted by all the related VMs.

For that, we will use [certstrap](https://github.com/square/certstrap/) tooling to simplify management of the certificates.
In the concourse job pipelines we will also interact with S3 using the script `s3init.sh`, so we must include `curl` and `ca-certificates`.

how to test:
-----------

Travis tests should execute the spec added in this PR and pass .
```
rake build:certstrap spec:certstrap
```

Notes
-----

A docker hub automated build must be setup with this Dockerfile.

The PR https://github.com/alphagov/paas-cf/pull/92 must be updated to use
the new container

Who?
----

Anyone but @keymon or @mtekel

### CODE REVIEW

File: certstrap/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ENV GIT_COMMIT 8b3cd66cfe2433e2e998c46a618eb09f732c2c2c
+ENV CERTSTRAP_VERSION 0.1.0
+
+RUN apk add --update git bash openssl curl ca-certificates \
+  && go get -d github.com/square/certstrap \
+  && cd ${GOPATH}/src/github.com/square/certstrap \
+  && git checkout ${GIT_COMMIT} \
+  && ./build \
+  && cp bin/certstrap /usr/local/bin \
+  && apk del git bash \
```
Comment (saliceti) [inline certstrap/Dockerfile]: You can add `&& rm -rf ${GOPATH} \`
(this will save 6MB...)

Comment (keymon) [inline certstrap/Dockerfile]: Don't be cheap!  :smile: 

I think we can change all these containers when we have our own CI system by a pipeline which compiles the binary statically and adds it to a naked alpine container.

**Comment by saliceti** [comment]
Tested successfully

File: certstrap/Dockerfile (hunk 1 above)
Comment (keymon) [inline certstrap/Dockerfile]: Done :kiss::dash:

**Pull request merged**